src/Aspire.Hosting/AspireStoreExtensions.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for <see cref="IDistributedApplicationBuilder"/> to create an <see cref="IAspireStore"/> instance.
+/// </summary>
+public static class AspireStoreExtensions
+{
+    internal const string AspireStorePathKeyName = "Aspire:Store:Path";
+
+    /// <summary>
+    /// Creates a new App Host store using the provided <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <returns>The <see cref="IAspireStore"/>.</returns>
+    public static IAspireStore CreateStore(this IDistributedApplicationBuilder builder)
```

Comment:
I know we can iterate on this API, so we don't have to block on this, but I think this is going to clutter up normal dev's intellisense. Normal apphost devs don't need to use this API, so I don't think it should be top-level like "AddContainer" or "AddPostgres", etc.

Reply:
Ok to move to `Azure.Hosting.ApplicationModel` instead?